projects/storefrontlib/cms-structure/seo/media-preconnect.service.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable } from '@angular/core';
+import { PageMetaLinkService } from './page-meta-link.service';
+import { WindowRef, FeatureToggles } from '@spartacus/core';
+import { MediaService } from '../../shared/components/media/media.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MediaPreconnectService {
+  protected pageMetaLinkService = inject(PageMetaLinkService);
+  protected mediaService = inject(MediaService);
+  private featureToggles = inject(FeatureToggles);
+  private windowRef = inject(WindowRef);
+
+  addPreconnectLink(): void {
+    if (!this.featureToggles.createMediaPreconnectLink) {
+      return;
+    }
+
+    const url = this.mediaService.getBaseUrl();
+    let domain: string | undefined;
+    try {
+      domain = new URL(url).origin;
+    } catch {
+      domain = undefined;
+    }
+
+    if (domain && domain !== this.windowRef.location.origin) {
```

Comment:
Let's explain the _reason_ of the code also here in the source code (not only in feature toggle's description -> because the feature toggle will be eventually removed in 12months from now)

```suggestion
    // Preconnecting is not needed (and won't be performed) if the domain of the media base url
    // is the same as the storefront's domain
    if (domain && domain !== this.windowRef.location.origin) {
```